src/Compilers/CSharp/Portable/BoundTree/OutDeconstructVarPendingInference.cs

```diff
@@ -22,5 +22,14 @@ public BoundDeconstructValuePlaceholder FailInference(Binder binder)
         {
             return SetInferredType(TypeSymbolWithAnnotations.Create(binder.CreateErrorType()), binder, success: false);
         }
+
+        protected override BoundExpression ShallowClone()
```

Comment:
>ShallowClone [](start = 43, length = 12)

:memo: I debated whether to throw here, since can't be reached at the moment. I could add a `Debug.Assert` so that we can detect when this ends up being exercised. What do you think?

Reply:
`throw ExceptionUtilities.Unreachable;`

---
In reply to: [251632000](https://github.com/dotnet/roslyn/pull/32850#discussion_r251632000) [](ancestors = 251632000)

Reply:
My concern with doing that is that we might need to use `ShallowClone` in some new scenario in the future. At that time, I'm not sure we would think to check whether the implementation is complete  (`OutDeconstructVarPendingInference`, `BoundQueryClause` both raise this problem).